demo/riskmodels/fraud_detection/spark_lgbm.py

```diff
@@ -0,0 +1,204 @@
+    from synapse.ml.lightgbm import LightGBMClassifier
+    model = LightGBMClassifier(isProvideTrainingMetric=True, 
+                               featuresCol='features', labelCol=label_col, 
+                               isUnbalance=True, 
+                               **model_params)
+    model = model.fit(train_dataset)
+    return model
+
+def evaluate(spark, test_result, label_col):
+    evaluator = BinaryClassificationEvaluator(labelCol=label_col, metricName="areaUnderROC")
+    auc = evaluator.evaluate(test_result)
+
+    label = np.array(predictions.select(label_col).collect()).reshape(-1).astype(np.float32)
+    prediction = np.array(predictions.select('probability').collect())[:, :, 1].reshape(-1)
+    ks = ks_2samp(label, prediction)
+    return auc, ks
+
+def write_dataset_to_s3(eval_dataset, eval_out_path, **kwargs):
+    start = time.time()
+    eval_dataset.write.parquet(eval_out_path, mode="overwrite")
+    print('Debug -- write_dataset_to_s3 cost time:', time.time() - start)
+
+def convert_model(lgbm_model: LGBMClassifier or Booster, input_size: int) -> bytes:
+    initial_types = [("input", FloatTensorType([-1, input_size]))]
+    onnx_model = convert_lightgbm(lgbm_model, initial_types=initial_types, target_opset = 9)
+    return onnx_model   
+
+def get_onnx_model(model, len_data_columns):
+    booster_model_str = model.getLightGBMBooster()
+    booster_model_str = booster_model_str.modelStr()
+    booster_model_str = booster_model_str.get()
+    booster = lgb.Booster(model_str = booster_model_str)
+    onnx_model = convert_model(booster, len_data_columns)
+    return onnx_model
+
+def save_onnx_model(onnx_model, model_onnx_path, **kwargs):
+    onnxmltools.utils.save_model(onnx_model,'lightgbm.onnx')
+    loaded_model = onnxmltools.utils.load_model('lightgbm.onnx')
+    subprocess.run(['aws', 's3', 'cp', 'lightgbm.onnx', model_onnx_path], cwd='./')
+    print(type(onnx_model))
+    print(type(loaded_model))
+    return loaded_model
+
+def load_onnx(onnx_ml):
+    onnx_ml = onnx_ml.setDeviceType("CPU") \
+                     .setFeedDict({"input": "features"}) \
+                     .setFetchDict({"probability": "probabilities", "prediction": "label"}) \
+                     .setMiniBatchSize(5000)
+    return onnx_ml
+
+def launch_onnx_model(loaded_model):
+    from synapse.ml.onnx import ONNXModel
+    onnx_ml = ONNXModel().setModelPayload(loaded_model.SerializeToString())
+    print("Model inputs:" + str(onnx_ml.getModelInputs()))
+    print("Model outputs:" + str(onnx_ml.getModelOutputs()))
+    print("Model type:" )
+    print(type(onnx_ml))
+    return onnx_ml
+
+def onnx_transform(onnx_ml, test_data):
+    result = onnx_ml.transform(test_data)
+    print("ONNX transform sample:", result.limit(10).toPandas())
+
+if __name__=="__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--conf', type=str, action='store', default='', help='config file path')
+    args = parser.parse_args()
```

Comment:
Can we put parser logic into load_config func ?

Reply:
I think it may be better parse python input parameters outside.